tracing/src/test/java/com/palantir/tracing/CloseableTracerTest.java

```diff
@@ -53,4 +58,26 @@ public void startsAndClosesSpanWithType() {
         }
         assertThat(Tracer.getAndClearTrace().top()).isEmpty();
     }
+
```

Comment:
I've added a test which uses the API, but it's not very good documentation since it defines the recorder inline where in practice a single recorder would be reused (although to be fair, the inner recorder is still far more efficient than the map codepath if objects aren't being reused)

Reply:
I updated the description of this PR describing the new abstraction, hopefully that's more helpful.